core/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -125,6 +132,14 @@ public KeywordFieldMapper build(BuilderContext context) {
                 } else if (propName.equals("eager_global_ordinals")) {
                     builder.eagerGlobalOrdinals(XContentMapValues.nodeBooleanValue(propNode));
                     iterator.remove();
+                } else if (propName.equals("analyzer")) {
+                    String analyzerName = propNode == null ? null : propNode.toString();
+                    NamedAnalyzer analyzer = analyzerName == null ? null : parserContext.getIndexAnalyzers().get(analyzerName);
+                    if (analyzerName != null && analyzer == null) {
```

Comment:
analyzerName should never be null here or there is something messed up with our parsing right?